mindmeld/auto_annotator.py

```diff
@@ -835,5 +839,300 @@ def valid_entity_check(self, entity):
         return Entity.is_system_entity(entity) or entity in self.supported_entity_types
 
 
+class Translator(ABC):
+    """Abstract Translator Base Class for Translators to be used by Mindmeld."""
+
+    def __init__(self):
+        """Creates a translation client after finding the credential path."""
+        self.translate_client = None
+
+    @abstractmethod
+    def get_translate_client(self):
+        """
+        Args:
+            text (str): Input text
+        Returns:
+            language_code (str): Detected Language Code
+        """
+        raise NotImplementedError("Subclasses must implement this method")
+
+    @abstractmethod
+    def detect_language(self, text):
+        """
+        Args:
+            text (str): Input text
+        Returns:
+            language_code (str): Detected Language Code
+        """
+        raise NotImplementedError("Subclasses must implement this method")
+
+    @abstractmethod
+    def translate(self, text, destination_language):
+        """
+        Args:
+            text (str): Input text
+            destination_language (str): Language code for language to translate the given text to.
+        Returns:
+            translated_text (str): Translated text
+        """
+        raise NotImplementedError("Subclasses must implement this method")
+
+
+class GoogleTranslator(Translator):
+    """Class for translation using the Google Translate API."""
+
+    def __init__(self):
+        """Initializes the translate_client."""
+        self.translate_client = self.get_translate_client()
+
+    def get_translate_client(self):
+        """Creates a translation client after finding the credential path."""
+        GoogleTranslator._check_credential_exists()
```

Comment:
Does this raise an exception if the creds are bad? If so, we should catch the exception and display proper error messages.